File: app/src/main/java/com/example/weatherapp/FcstService.java
package com.example.weatherapp;

import io.reactivex.rxjava3.core.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface FcstService {
    @GET("getVilageFcst")
    Observable<FcstResult> getVilageFcst(@Query("serviceKey") String serviceKey,
                                         @Query("pageNo") String pageNo,
                                         @Query("numOfRows") String numOfRows,
                                         @Query("dataType") String dataType,
                                         @Query("base_date") String base_date,
                                         @Query("base_time") String base_time,
                                         @Query("nx") String nx,
                                         @Query("ny") String ny);
    @GET("getUltraSrtFcst")
    Observable<FcstResult> getUltraSrtFcst(@Query("serviceKey") String serviceKey,
                                        @Query("pageNo") String pageNo,
                                        @Query("numOfRows") String numOfRows,
                                        @Query("dataType") String dataType,
                                        @Query("base_date") String base_date,
                                        @Query("base_time") String base_time,
                                        @Query("nx") String nx,
                                        @Query("ny") String ny);
}

interface ArpltnService {
    @GET("getMsrstnAcctoRltmMesureDnsty")
    Observable<ArpltnResult> getDnsty(@Query("serviceKey") String serviceKey,
                          @Query("returnType") String returnType,
                          @Query("numOfRows") String numOfRows,
                          @Query("pageNo") String pageNo,
                          @Query("dataTerm") String dataTerm,
                          @Query("ver") String ver,
                          @Query("stationName") String stationName);
}

File: app/src/main/java/com/example/weatherapp/data/source/remote/fcst/FcstService.kt
package com.example.weatherapp.data.source.remote.fcst

import retrofit2.http.GET
import com.example.weatherapp.data.source.remote.fcst.FcstResult
import io.reactivex.rxjava3.core.Observable
import retrofit2.http.Query

interface FcstService {
    @GET("getVilageFcst")
    fun getVilageFcst(
        @Query("serviceKey") serviceKey: String?,
        @Query("pageNo") pageNo: String?,
        @Query("numOfRows") numOfRows: String?,
        @Query("dataType") dataType: String?,
        @Query("base_date") base_date: String?,
        @Query("base_time") base_time: String?,
        @Query("nx") x: Int,
        @Query("ny") y: Int
    ): Observable<FcstResult>

    @GET("getUltraSrtFcst")
    fun getUltraSrtFcst(
        @Query("serviceKey") serviceKey: String?,
        @Query("pageNo") pageNo: String?,
        @Query("numOfRows") numOfRows: String?,
        @Query("dataType") dataType: String?,
        @Query("base_date") base_date: String?,
        @Query("base_time") base_time: String?,
        @Query("nx") x: Int,
        @Query("ny") y: Int
    ): Observable<FcstResult>
}